### PR TITLE
Add /block endpoint (without transactions)

### PR DESCRIPTION
**Pull request by nddeluca**
Adds the block endpoint.  Transactions are not yet returned, but all other required block data is.

Blocks can be queried by index or by height, and the genesis block sets it's parent to itself according to Rosetta requirements.

Request via index:
```
curl -X 'POST' \
  'http://localhost:8000/block' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "network_identifier": {
    "blockchain": "Kava",
    "network": "kava-7"
  },
  "block_identifier": {
    "index": 100
  }
}'
```

Request via hash:
```
curl -X 'POST' \
  'http://localhost:8000/block' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "network_identifier": {
    "blockchain": "Kava",
    "network": "kava-7"
  },
  "block_identifier": {
    "hash": "D92BDF0B5EDB04434B398A59B2FD4ED3D52B4820A18DAC7311EBDF5D37467E75"
  }
}'
```

Response (it's the same block in both request examples):
```
{
  "block": {
    "block_identifier": {
      "index": 100,
      "hash": "D92BDF0B5EDB04434B398A59B2FD4ED3D52B4820A18DAC7311EBDF5D37467E75"
    },
    "parent_block_identifier": {
      "index": 99,
      "hash": "8EA67B6F7927DB941F86501D1757AC6804C1D21B7A75B9DA3F16A3C81C397E50"
    },
    "timestamp": 1617894805837,
    "transactions": []
  }
}
```

### CODE REVIEW

File: kava/client.go [hunk 1]
```diff
@@ -176,3 +164,56 @@ func (c *Client) Balance(
...
+			Timestamp:             block.Block.Header.Time.UnixNano() / int64(1e6),
+			Transactions:          []*types.Transaction{},
+		},
+	}, nil
+}
+
+func (c *Client) getBlockResult(blockIdentifier *types.PartialBlockIdentifier) (block *ctypes.ResultBlock, err error) {
+	switch {
+	case blockIdentifier == nil:
+		block, err = c.rpc.Block(nil)
```
Comment (karzak) [inline kava/client.go:207]: Just curiousity, what does passing nil to `rpc.Block` result in?

**Review by karzak — APPROVED** [review]
LGTM

File: kava/client.go (hunk 1 above)
Comment (nddeluca) [reply kava/client.go:207]: It'll return the latest height. It's a requirement for the /account/balance endpoint, not for the /block endpoint.

Rpc reference is here https://github.com/tendermint/tendermint/blob/master/rpc/core/blocks.go#L92

Comment (nddeluca) [reply kava/client.go:210]: Added some comments to make that clear in the code

**Pull request merged**